src/library/autodj/autodjfeature.cpp

```diff
@@ -293,32 +306,31 @@ void AutoDJFeature::constructCrateChildModel() {
 void AutoDJFeature::onRightClickChild(const QPoint& globalPos,
                                       QModelIndex index) {
     TreeItem* pClickedItem = static_cast<TreeItem*>(index.internalPointer());
+    m_pMenu->clear();
     if (m_pCratesTreeItem == pClickedItem) {
         // The "Crates" parent item was right-clicked.
         // Bring up the context menu.
-        QMenu crateMenu;
-        crateMenu.setTitle(tr("Add Crate as Track Source"));
+        m_pCrateMenu->clear();
+        m_pCrateMenu->setTitle(tr("Add Crate as Track Source"));
         CrateSelectResult nonAutoDjCrates(m_pTrackCollection->crates().selectAutoDjCrates(false));
         Crate crate;
         while (nonAutoDjCrates.populateNext(&crate)) {
-            auto pAction = std::make_unique<QAction>(crate.getName(), &crateMenu);
+            auto pAction = std::make_unique<QAction>(crate.getName(), m_pCrateMenu);
             m_crateMapper.setMapping(pAction.get(), crate.getId().value());
             connect(pAction.get(),
                     &QAction::triggered,
                     [=](bool) { m_crateMapper.map(); });
             crateMenu.addAction(pAction.get());
             pAction.release();
         }
-        QMenu contextMenu;
-        contextMenu.addMenu(&crateMenu);
-        contextMenu.exec(globalPos);
+        m_pMenu->addMenu(m_pCrateMenu);
```

Comment:
change ```m_pMenu``` to ```m_pCrateMenu```

Reply:
and there are more build errors in ```library.c``` around line 132 "BrowseFeature"

Reply:
for AutoDJ `crateMenu` leftover was the issue.
`m_pMenu` and `m_pCrateMenu` are correct

`m_pBrowseFeature` is also fixed now

Reply:
thx!